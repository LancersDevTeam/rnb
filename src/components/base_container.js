// @flow
import React, { Component } from 'react'
import { StatusBar } from 'react-native'
import { Container as NBContainer } from 'native-base'
import styles from './base_container_style'

type State = any
type Props = any

export class BaseContainer extends Component<Props, State> {
  render() {
    return (
      <NBContainer {...this.props} style={styles.container}>
        <StatusBar />
        {this.props.children}
      </NBContainer>
    )
  }
}
