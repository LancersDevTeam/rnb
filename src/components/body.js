// @flow
import React, { Component } from 'react'
import { Body as NBBody } from 'native-base'

type State = any
type Props = any

export class Body extends Component<Props, State> {
  render() {
    return <NBBody {...this.props}>{this.props.children}</NBBody>
  }
}
