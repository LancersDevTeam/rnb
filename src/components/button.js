// @flow
import React, { Component } from 'react'

import { Button as NBButton } from 'native-base'

type State = any
type Props = any

export class Button extends Component<Props, State> {
  render() {
    return (
      <NBButton rounded block {...this.props}>
        {this.props.children}
      </NBButton>
    )
  }
}
