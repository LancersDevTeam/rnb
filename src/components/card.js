// @flow
import React, { Component } from 'react'
import { Card as NBCard } from 'native-base'

type State = any
type Props = any

export class Card extends Component<Props, State> {
  render() {
    return <NBCard {...this.props}>{this.props.children}</NBCard>
  }
}
