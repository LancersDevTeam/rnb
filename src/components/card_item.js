// @flow
import React, { Component } from 'react'

import { CardItem as NBCardItem } from 'native-base'

type State = any
type Props = any

export class CardItem extends Component<Props, State> {
  _root: any

  setNativeProps = (nativeProps: any) => {
    this._root.setNativeProps(nativeProps)
  }

  render() {
    return (
      <NBCardItem ref={component => (this._root = component)} {...this.props}>
        {this.props.children}
      </NBCardItem>
    )
  }
}
