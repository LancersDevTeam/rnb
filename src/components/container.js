// @flow
import React, { Component } from 'react'
import { Container as NBContainer } from 'native-base'

type State = any
type Props = any

export class Container extends Component<Props, State> {
  render() {
    return <NBContainer {...this.props}>{this.props.children}</NBContainer>
  }
}
