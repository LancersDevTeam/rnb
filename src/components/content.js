// @flow
import React, { Component } from 'react'
import { Content as NBContent } from 'native-base'

type State = any
type Props = any

export class Content extends Component<Props, State> {
  render() {
    return <NBContent {...this.props}>{this.props.children}</NBContent>
  }
}
