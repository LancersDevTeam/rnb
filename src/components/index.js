export * from './base_container'
export * from './body'
export * from './button'
export * from './card'
export * from './card_item'
export * from './container'
export * from './content'
export * from './deck_swiper'
export * from './form'
export * from './header'
export * from './icon'
export * from './input'
export * from './input_group'
export * from './item'
export * from './label'
export * from './left'
export * from './list'
export * from './list_item'
export * from './right'
export * from './separator'
export * from './switch'
export * from './text'
export * from './thumbnail'
