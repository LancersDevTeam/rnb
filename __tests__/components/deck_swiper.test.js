// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import { Body, Card, CardItem, Left, Text } from 'native-base'
import { DeckSwiper } from 'app/components'

const cards = [
  {
    text: 'Card One',
    name: 'One'
  }
]

it('renders correctly', () => {
  const tree = renderer
    .create(
      <DeckSwiper
        dataSource={cards}
        renderItem={item => (
          <Card style={{ elevation: 3 }}>
            <CardItem>
              <Left>
                <Body>
                  <Text>{item.text}</Text>
                  <Text note>NativeBase</Text>
                </Body>
              </Left>
            </CardItem>
            <CardItem cardBody>
              <Text>dummy</Text>
            </CardItem>
            <CardItem>
              <Text>{item.name}</Text>
            </CardItem>
          </Card>
        )}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
