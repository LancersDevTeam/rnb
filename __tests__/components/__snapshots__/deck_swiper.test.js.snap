// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
      "position": "relative",
    }
  }
>
  <View>
    <View
      collapsable={undefined}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "left": 0,
          "opacity": 0.8,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "scale": 0.8,
            },
          ],
        }
      }
    />
    <View
      collapsable={undefined}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
            Object {
              "rotate": "0deg",
            },
          ],
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderBottomWidth": 0.5,
              "borderColor": "#ccc",
              "borderLeftWidth": 0.5,
              "borderRadius": 2,
              "borderRightWidth": 0.5,
              "borderTopWidth": 0.5,
              "elevation": 3,
              "flex": 1,
              "flexWrap": "nowrap",
              "marginBottom": 5,
              "marginLeft": 2,
              "marginRight": 2,
              "marginTop": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 1.5,
            },
            Object {
              "elevation": 3,
            },
          ]
        }
        virtual={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            }
          }
          virtual={undefined}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flex": 1,
                "flexDirection": "row",
              }
            }
            virtual={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": null,
                  "alignSelf": "center",
                  "flex": 1,
                  "marginLeft": 10,
                }
              }
              virtual={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  }
                }
                uppercase={false}
                virtual={undefined}
              >
                Card One
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                note={true}
                style={
                  Object {
                    "color": "#808080",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "marginRight": 20,
                  }
                }
                uppercase={false}
                virtual={undefined}
              >
                NativeBase
              </Text>
            </View>
          </View>
        </View>
        <View
          cardBody={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": -5,
              "paddingLeft": -5,
              "paddingRight": -5,
              "paddingTop": -5,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
                "marginTop": 5,
              }
            }
            uppercase={false}
            virtual={undefined}
          >
            dummy
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              }
            }
            uppercase={false}
            virtual={undefined}
          >
            One
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
